cmd_stream.go

```diff
@@ -300,19 +301,37 @@ func (m *Miniredis) makeCmdXrange(reverse bool) server.Cmd {
 
 // XGROUP
 func (m *Miniredis) cmdXgroup(c *server.Peer, cmd string, args []string) {
-	if (len(args) == 4 || len(args) == 5) && strings.ToUpper(args[0]) == "CREATE" {
+	subCmd, args := strings.ToUpper(args[0]), args[1:]
+	switch subCmd {
+	case "CREATE":
 		m.cmdXgroupCreate(c, cmd, args)
-	} else {
-		j := strings.Join(args, " ")
-		err := fmt.Sprintf("ERR 'XGROUP %s' not supported", j)
+	case "DESTROY":
+		m.cmdXgroupDestroy(c, cmd, args)
+	case "CREATECONSUMER":
+		m.cmdXgroupCreateconsumer(c, cmd, args)
+	case "DELCONSUMER":
+		m.cmdXgroupDelconsumer(c, cmd, args)
+	case "HELP":
+	case "SETID":
+		err := fmt.Sprintf("ERR 'XGROUP %s' not supported", subCmd)
 		setDirty(c)
 		c.WriteError(err)
+	default:
+		setDirty(c)
+		c.WriteError(fmt.Sprintf(
+			"ERR Unknown subcommand or wrong number of arguments for '%s'. Try XGROUP HELP.",
+			subCmd,
+		))
 	}
 }
 
 // XGROUP CREATE
 func (m *Miniredis) cmdXgroupCreate(c *server.Peer, cmd string, args []string) {
-	stream, group, id := args[1], args[2], args[3]
+	if len(args) != 3 && len(args) != 4 {
+		c.WriteError(errWrongNumber("CREATE"))
```

Comment:
All errors also need a `setDirty(c)` (for transactions).

Reply:
Got it–fixed!